runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/QueryParameters.kt

```diff
@@ -49,6 +49,8 @@ public fun Map<String, String>.toQueryParameters(): QueryParameters {
 
 private class QueryParametersImpl(values: Map<String, List<String>> = emptyMap()) : QueryParameters, StringValuesMapImpl(true, values) {
     override fun toString(): String = "QueryParameters ${entries()}"
+
+    override fun equals(other: Any?): Boolean = other is QueryParameters && entries() == other.entries()
```

Comment:
why are we defining equality for this? I'm not sure this does what we expect since query params can use an internal case insensitive map that _does not_ keep the uses the original given key when returning the entry set.

Reply:
It was added to support the `assertEquals` checks in the new parser test (to check equality on a full URL). The alternative would be to inspect individual members in those tests (or adjust the equality check if we want it and feel this implementation isn't adequate).